### DISCUSSION TITLE
Call api in build time

### DISCUSSION THREAD

**Discussion by MohammadMiras** [vercel/next.js#47975] [Help]
### Summary

We have problem with communication with api at build time

Hi  We are a programming team from Iran who are developing the website with the help of Next 13, but due to the lack of stable internet access during  build time we sometimes encounter problems. We found that these problems are due to the inability Contact with api  occurs at build time. Is it possible in Next 13 not to make this call during build?

_No response_

**Comment by Abdullah-moiz** [comment]
1. One potential solution would be to mock the API response during the build process. This would involve creating a fake response that the build process can use instead of making an actual API call. This can help to avoid errors caused by unstable internet access.

2.  You may want to consider caching the API response during the build process. This involves storing the response from the API in a cache so that it can be accessed later, even if there is no internet connection during the build.



you can consider following libraries for mockup or cache
- [github.com/mswjs/msw](https://github.com/mswjs/msw)
- [github.com/nock/nock](https://github.com/nock/nock)
- [github.com/node-cache/node-cache](https://github.com/node-cache/node-cache)

**Comment by MohammadMiras** [comment]
@Abdullah-moiz  how can I mock the API response during the build process?

I mean, is there a flag or something that I can use to know that I'm in the middle of a build?

I'm using Axios and a typical API call would be `axios.get(url).then(response => {}).catch(error => {})`. What code should I write for this purpose?

**Reply by Abdullah-moiz** [reply]
I mentioned a library that you can use for mocking, read the documentation to understand how can you mock 

Here is one basic example

```
If(process.env.NODE_ENV === 'production') {
  // Mock the API response for the build process
  mock.onGet('/api/data').reply(200, { message: 'Mock data' });
} else {
  // Use the real API response for the development process
  api.get('/api/data').then(response => {
    console.log(response.data);
  }).catch(error => {
    console.error(error);
  });
}
```

**Reply by umashankar-pardhi-amla** [reply]
But why this is required we are preparing build why we need to call API
